converters/fromZigbee.js

```diff
@@ -2572,35 +2572,38 @@ const converters = {
                 result[postfixWithEndpointName('pi_heating_demand', msg, model)] =
                     precisionRound(msg.data['pIHeatingDemand'], 0);
             }
-            if (typeof msg.data[0x4000] == 'number') {
-                result[postfixWithEndpointName('window_open_internal', msg, model)] = (msg.data[0x4000]);
+            if (msg.data.hasOwnProperty('danfossWindowOpenInternal')) {
+                result[postfixWithEndpointName('window_open_internal', msg, model)] = msg.data['danfossWindowOpenInternal'];
```

Comment:
This maps to 
0x00: Quarantine
0x01: Windows are closed
0x02: Hold ,Windows are maybe about to open
0x03: Open window detected
0x04: In window open state from external

Right? We can probably use a nice enum here! (https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/lib/constants.js already has a bunch so adding a danfossWindowOpen or something.

Here is an example of how the mapping is done:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/50ad929821ff1a93dd3e826f2a4ed4fa078d8d6c/converters/fromZigbee.js#L144
https://github.com/Koenkk/zigbee-herdsman-converters/blob/50ad929821ff1a93dd3e826f2a4ed4fa078d8d6c/converters/toZigbee.js#L1118

Reply:
Now that I am awake... it would be clean to add thes to constants.js I think.

```
const danfossWindowOpen = {
    0: 'quarantine',
    1: 'closing',
    2: 'hold',
    3: 'open',
    4: 'external_open',
};
```

Then in fromZigbee:
```
result[postfixWithEndpointName('window_open_internal', msg, model)]  = constants.danfossWindowOpen.hasOwnProperty(msg.data['danfossWindowOpenInternal']) ?
                    constants.danfossWindowOpen[msg.data['danfossWindowOpenInternal']] : msg.data['danfossWindowOpen'];
```

There is no toZigbee for danfossWindowOpenInternal so nothing needs to be done there.

That should make it so people don't have to look at the PDF/source to know that 3 and 4 are both open but the later because it was forced etc...

Reply:
Thanks for the feedback and telling me the enum option. https://github.com/Koenkk/zigbee-herdsman-converters/pull/2592/commits/1fc978fd5ba3e478039c1e98662ed9c2f39bd51a

constans.js - i have decided to add the attribute with the rest of the thermostat ones rather than at the bottom of the list, I have debated it for a while and chose this option
fromZigbee.js - your "else" key is not correct but it doesn't matter :). Also, I hate the formatting of the ternary I had to do in my code but Lint didn't like it my way (despite me thinking is much more readable). This is how I would have done it
![image](https://user-images.githubusercontent.com/9019306/119203652-7882f180-ba8b-11eb-81ed-3d9ee16af0c0.png)

I guess this resolves this too :)